src/gateway/opcodes/LazyRequest.ts

```diff
@@ -212,8 +212,49 @@ export async function onLazyRequest(this: WebSocket, { d }: Payload) {
 	// TODO: check data
 	check.call(this, LazyRequestSchema, d);
 	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	const { guild_id, typing, channels, activities } = d as LazyRequestSchema;
-	if (!channels) return; // TODO: new clients use this to request PRESENCE_UPDATE for `members`
+	const { guild_id, typing, channels, activities, members } =
+		d as LazyRequestSchema;
+
+	if (members) {
+		// Client has requested a PRESENCE_UPDATE for specific member
+
+		await Promise.all([
+			members.map(async (x) => {
+				if (!x) return;
+				const didSubscribe = await subscribeToMemberEvents.call(
+					this,
+					x,
+				);
+				if (!didSubscribe) return;
+
+				// if we didn't subscribe just now, this is a new subscription
+				// and we should send a PRESENCE_UPDATE immediately
+
+				const sessions = await Session.find({ where: { user_id: x } });
+				const session = getMostRelevantSession(sessions);
+				// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+				// @ts-ignore
+				if (session?.status == "unknown") session.status = "online";
+				const user = (await User.getPublicUser(x)).toPublicUser(); // why is this needed?
+
+				return Send(this, {
+					op: OPCODES.Dispatch,
+					s: this.sequence++,
+					t: "PRESENCE_UPDATE",
+					d: {
+						user: user,
+						activities: session?.activities || [],
+						client_status: session?.client_info,
+						status: session?.status || "offline",
+					} as Presence,
+				});
+			}),
+		]);
+
+		if (!channels) return;
+	}
+
+	if (!channels) throw new Error("Must provide channel ranges");
```

Comment:
check the presence right of presence sender (yes, the right way around)

Reply:
The above code is actually pretty useless, since the client will only request members in the member list, and since they're in the member list you're already subscribed to them. I'm not sure what the point of lazy request members is, on discord.com now.

Reply:
third party clients and bots can request them

Reply:
This does not address what I said. What is the purpose of it *on Discord.com*